### PR TITLE
Extract strings in code to resources/values/strings.xml and used it from there

### PR DESCRIPTION
**Pull request by b0r1ngx**
Hello, Arkadii, I'm done with extracting user strings from code, please review.

I think that we may extract strings only that user interact with, because later its valueble for translate it for example.

## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic text retrieval for various UI elements using resource strings.
- **Refactor**
	- Refactored UI components to use resource strings for actions, labels, and buttons.
- **Documentation**
	- Added a TODO comment in the `Main.kt` file for potential resource integration with WebAssembly.

### CODE REVIEW

File: composeApp/src/commonMain/composeResources/values/strings.xml [hunk 1]
```diff
@@ -1,2 +1,22 @@
...
+    <string name="mine_count">Mine count</string>
+    <string name="width">Width</string>
+    <string name="height">Height</string>
+    <string name="restart">Restart</string>
+    <string name="settings">Settings</string>
+
+    // TODO: Check how Accessibility work, if its valueble - change trailing for: %d mines left.
+    // TODO: Accessibility check gives info that user can't select @Composable Counter`s with tab-navigation in our app - need to undertsand why and fix it.
+    <string name="mines_counter">Counter of remaining mines, mines left: %d</string>
+    <string name="timer">Timer is on %d</string>
```
Comment (arkivanov) [inline composeApp/src/commonMain/composeResources/values/strings.xml]: Maybe let's write something like "Duration: %d seconds"?

File: composeApp/src/commonMain/composeResources/values/strings.xml [hunk 2]
```diff
@@ -1,2 +1,22 @@
...
+    <string name="cancel">Cancel</string>
+    <string name="mine_count">Mine count</string>
+    <string name="width">Width</string>
+    <string name="height">Height</string>
+    <string name="restart">Restart</string>
+    <string name="settings">Settings</string>
+
+    // TODO: Check how Accessibility work, if its valueble - change trailing for: %d mines left.
+    // TODO: Accessibility check gives info that user can't select @Composable Counter`s with tab-navigation in our app - need to undertsand why and fix it.
+    <string name="mines_counter">Counter of remaining mines, mines left: %d</string>
```
Comment (arkivanov) [inline composeApp/src/commonMain/composeResources/values/strings.xml]: Since this is for accessibility, maybe find a shorter text? E.g. "Mines left: %d".

Comment (arkivanov) [reply composeApp/src/commonMain/composeResources/values/strings.xml]: Also, let's add a prefix for all accessibility strings, e.g. `desc_mines_left`.

File: composeApp/src/commonMain/composeResources/values/strings.xml (hunk 1 above)
Comment (arkivanov) [reply composeApp/src/commonMain/composeResources/values/strings.xml]: Also let's name it `desc_duration`.

File: composeApp/src/commonMain/composeResources/values/strings.xml [hunk 3]
```diff
@@ -1,2 +1,22 @@
...
+    <string name="apply">Apply</string>
+    <string name="cancel">Cancel</string>
+    <string name="mine_count">Mine count</string>
+    <string name="width">Width</string>
+    <string name="height">Height</string>
+    <string name="restart">Restart</string>
+    <string name="settings">Settings</string>
+
+    // TODO: Check how Accessibility work, if its valueble - change trailing for: %d mines left.
+    // TODO: Accessibility check gives info that user can't select @Composable Counter`s with tab-navigation in our app - need to undertsand why and fix it.
```
Comment (arkivanov) [inline composeApp/src/commonMain/composeResources/values/strings.xml]: Could you please elaborate on this comment? What does it mean?

File: composeApp/src/wasmJsMain/kotlin/Main.kt [hunk 4]
```diff
@@ -33,6 +33,7 @@ fun main() {
             null
         }
 
+    // TODO: Understand how to use resources for WAsm
```
Comment (arkivanov) [inline composeApp/src/wasmJsMain/kotlin/Main.kt]: Could you please elaborate? I thought resources are working the same way on wasm.

**Review by arkivanov — CHANGES_REQUESTED** [review]

File: composeApp/src/commonMain/composeResources/values/strings.xml (hunk 3 above)
Comment (b0r1ngx) [reply composeApp/src/commonMain/composeResources/values/strings.xml]: Its more easy to show, check this out: https://youtu.be/_QhpFqwDvp0

You can also try VoiceOver on your macOS system: press cmd + F5 (mic button).
When i pressing tab in app, VoiceOver navigating us per elements, u can hear:
Settings button
Restart button
unknown <- its element of game field of mines (one more idea: user can access each cell with tab)

And we doesn't navigated (when pressing `tab` button) to Counter's and not hear information that it has.

File: composeApp/src/wasmJsMain/kotlin/Main.kt (hunk 4 above)
Comment (b0r1ngx) [reply composeApp/src/wasmJsMain/kotlin/Main.kt]: I just not get in touch, how to get resources when we not in composable context, only using coroutines, like in this guide? (https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-images-resources.html#strings)

**Comment by b0r1ngx** [comment]
One more thing that I want to say, the task of increasing level of accessibility is a not easy task and requires additional work and tests, so I think in context of this PR we may not target on this, but awesome that we have discussion about it!

File: composeApp/src/wasmJsMain/kotlin/Main.kt (hunk 4 above)
Comment (b0r1ngx) [reply composeApp/src/wasmJsMain/kotlin/Main.kt]: We can try do it like this:
```Kotlin
    CanvasBasedWindow(title = Res.string.app_name.key, canvasElementId = "ComposeTarget") {
        RootContent(root)
    }
```

Comment (b0r1ngx) [reply composeApp/src/wasmJsMain/kotlin/Main.kt]: > We can try do it like this:
> 
> ```kotlin
>     CanvasBasedWindow(title = Res.string.app_name.key, canvasElementId = "ComposeTarget") {
>         RootContent(root)
>     }
> ```

Nope. it gives a key of itself, for this snippet, it returns: "app_name"

Comment (arkivanov) [reply composeApp/src/wasmJsMain/kotlin/Main.kt]: I think for now we can only use resources in Composable functions. See https://youtrack.jetbrains.com/issue/KT-49981 for a general resources API.

File: composeApp/src/commonMain/composeResources/values/strings.xml (hunk 3 above)
Comment (arkivanov) [reply composeApp/src/commonMain/composeResources/values/strings.xml]: This makes sense, but should this comment be here? Let's just remove it for now. We could just follow-up with a fix at some point later.

File: composeApp/src/wasmJsMain/kotlin/Main.kt (hunk 4 above)
Comment (arkivanov) [reply composeApp/src/wasmJsMain/kotlin/Main.kt]: Let's write something like:

```
// TODO: Take the title from resources after https://youtrack.jetbrains.com/issue/KT-49981
```

**Review by arkivanov — DISMISSED** [review]

**Comment by b0r1ngx** [comment]
Is `GIVEN_created_WHEN_first_click_on_cell_THEN_state_started` test start failing?
(check auto build)

**Comment by arkivanov** [comment]
> Is `GIVEN_created_WHEN_first_click_on_cell_THEN_state_started` test start failing? (check auto build)

I think test test is flaky, filed #22.

**Review by arkivanov — APPROVED** [review]

**Pull request merged**